app/src/main/java/com/github/sdp_begreen/begreen/fragments/AdviceFragment.kt

```diff
@@ -0,0 +1,70 @@
+package com.github.sdp_begreen.begreen.fragments
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.fragment.app.Fragment
+import com.github.sdp_begreen.begreen.R
+
+// TODO: Rename parameter arguments, choose names that match
+// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+private const val ARG_PARAM1 = "param1"
```

Comment:
Overall, the code looks good and well-structured. However, I have a few comments:

The private properties param1 and param2 are not used anywhere except in the onCreate method. If they are not used anywhere else, they can be made local variables in the onCreate method.

In the onCreateView method, you can use the Elvis operator (?:) instead of orEmpty() to provide a default value when the parameter is null.

Other than these minor points, the code looks good to me. Great job!

Reply:
Thanks for your comments,

For the first one, actually param1 and param 2 are used in multiple methods, they are initialised in the method `onCreate` by getting their value from the `arguments`, and then they are used in the method `onCreateView`, so I cannot have them private to the `onCreate` method.

Or did I understood your comment wrong, do you think I should write it differently ?

And for the second comment, you are right, I can also use the Elvis operator, but that is actually what the `orEmpty()` method does under the hood, it will check that the string is not null, and if it is it will simply return the empty string (i.e. "").  So it would be equivalent to `param1?.let { " $it" } ?: ""`. But you are right I could have written it like that, thanks for the tips.